sdk-base/src/main/java/com/mapbox/maps/plugin/viewport/experimental/ViewportStatus.kt

```diff
@@ -0,0 +1,60 @@
+package com.mapbox.maps.plugin.viewport.experimental
+
+import com.mapbox.maps.plugin.viewport.experimental.state.ViewportState
+import com.mapbox.maps.plugin.viewport.experimental.transition.ViewportTransition
+
+/**
+ * Represents the status of the viewport.
+ *
+ * It could be either a [ViewportState] or [ViewportTransition].
+ */
+sealed class ViewportStatus {
+  /**
+   * Represents the current status is a [ViewportState].
+   *
+   * The state is null if current status is IDLE.
```

Comment:
If sealed class is used - maybe extract this to IDLE object ? 

```
sealed class ViewportStatus {
  object Idle : ViewportStatus()
  
  class State(...)
}
```

Reply:
The IDLE state is special and doesn't contain any transition or camera updates. 
The current abstraction makes the codebase more clean and clearly indicates that there's no state in action.

Reply:
> The IDLE state is special and doesn't contain any transition or camera updates.

Right, but won't using `ViewportStatus.Idle` instead of `ViewportStatus.State(null)` be even more specific? I see couple of benefits here : 
- From the name of the type itself user will understand that the state is idle, no animation or state is in action. Numerous comments about the `ViewportStatus.State(null)` will become unnecessary, hence code is self-documented with type name
- No way to make a mistake and use nullable idle state as `getStatus.status!!`. The only way to get status is to check type first

Reply:
Good points, will try it out and see how it changes our codebase
cc @macdrevx

Reply:
Just checked the codebase, if we add the Idle separate from State, we will have issues representing the fromState when we `setTransition`/`getTransition`, as we can't represent Idle as a ViewportState.

In most of our interface logics, we treat Idle a special ViewportState but not something separate from a state, that's why we used ViewportState? = null to represent Idle.